src/librustc_mir/build/expr/as_rvalue.rs

```diff
@@ -125,35 +125,36 @@ impl<'a,'tcx> Builder<'a,'tcx> {
                 //     to the same MIR as `let x = ();`.
 
                 // first process the set of fields
-                let fields: Vec<_> =
-                    fields.into_iter()
-                          .map(|f| unpack!(block = this.as_operand(block, f)))
-                          .collect();
+                let fields: Vec<_> = fields.into_iter()
+                                           .map(|f| unpack!(block = this.as_operand(block, f)))
+                                           .collect();
 
                 block.and(Rvalue::Aggregate(AggregateKind::Vec, fields))
             }
             ExprKind::Tuple { fields } => { // see (*) above
                 // first process the set of fields
-                let fields: Vec<_> =
-                    fields.into_iter()
-                          .map(|f| unpack!(block = this.as_operand(block, f)))
-                          .collect();
+                let fields: Vec<_> = fields.into_iter()
+                                           .map(|f| unpack!(block = this.as_operand(block, f)))
+                                           .collect();
 
                 block.and(Rvalue::Aggregate(AggregateKind::Tuple, fields))
             }
             ExprKind::Closure { closure_id, substs, upvars } => { // see (*) above
-                let upvars =
-                    upvars.into_iter()
-                          .map(|upvar| unpack!(block = this.as_operand(block, upvar)))
-                          .collect();
+                let upvars = upvars.into_iter()
+                                   .map(|upvar| unpack!(block = this.as_operand(block, upvar)))
+                                   .collect();
                 block.and(Rvalue::Aggregate(AggregateKind::Closure(closure_id, substs), upvars))
             }
             ExprKind::Adt { adt_def, variant_index, substs, fields, base } => { // see (*) above
                 // first process the set of fields
-                let fields_map: FnvHashMap<_, _> =
-                    fields.into_iter()
-                          .map(|f| (f.name, unpack!(block = this.as_operand(block, f.expr))))
-                          .collect();
+                let fields_map: FnvHashMap<_, _> = fields.into_iter()
```

Comment:
@nrc so I was looking more closely here...I would say this is a definite regression in readability. This is a tough heuristic (non-local), but certainly when I choose where to put a new line, I have a kind of "global optimization" over the entire function call chain going on, and I feel like it applies here. Is there any kind of open bug on this scenario? Do you even see what I'm pointing out :)